### PR TITLE
Add Nix support

### PR DESCRIPTION
**Pull request by mvkvc**
Changed my laptop to use Nix and added the config file `flake.nix` so that it works for me. Doesn't change anything for non-users so thought I would just open a PR, either way it's useful for me. @BrooklinJazz

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          ];
+          shellHook = ''
+              mkdir -p .nix_mix
+              mkdir -p .nix_hex
+              export MIX_HOME=$PWD/utils/.nix_mix
+              export HEX_HOME=$PWD/utils/.nix_hex
+              export PATH=$MIX_HOME/bin:$PATH
+              export PATH=$HEX_HOME/bin:$PATH
+              export PATH=$MIX_HOME/escripts:$PATH
+              export LIVEBOOK_DEFAULT_RUNTIME=mix:$PWD/utils
```
Comment (BrooklinJazz) [inline flake.nix:25]: Does this cause issues with installing additional dependencies?

Many Livebooks install extra projects in the setup section, and if we use a default runtime I don't think we can install those dependencies.

**Review by BrooklinJazz — APPROVED** [review]
I have not used Nix, so I'm happy to merge this in if it helps folks using Nix. I had one question before we merge it though - see my review.

File: flake.nix (hunk 1 above)
Comment (mvkvc) [reply flake.nix:25]: I set the default runtime for my convenience but seems like its not a good fit for this, I'll remove it so theres no confusion.

**Comment by mvkvc** [comment]
Sorry for the long list of commits, couldn't figure out if I can squash commits already pushed to the PR. I removed the non-default env variables as you asked.

**Comment by BrooklinJazz** [comment]
🥳🥳🥳🥳

**Pull request merged**